File: src/main/java/com/angorasix/projects/core/domain/project/Requirement.java
package com.angorasix.projects.core.domain.project;

import com.angorasix.projects.core.domain.Attribute;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * Value Object.
 * 
 * <p>Attribute of something, that conforms a requirement of a Project.
 * 
 * @author rozagerardo
 *
 */
@Getter
@AllArgsConstructor
public final class Requirement<P> {

  private final String article;
  private final Attribute<P> attribute;

}

File: src/main/kotlin/com/angorasix/projects/core/domain/project/Requirement.kt
package com.angorasix.projects.core.domain.project

import com.angorasix.projects.core.domain.Attribute

/**
 * Value Object.
 *
 * Attribute of something, that conforms a requirement of a Project.
 *
 * @author rozagerardo
 */
data class Requirement<P>(val article: String?,
                          val attribute: Attribute<P>?) {}